ada_2_fhir-r4/zibs2020/payload/0.0/zib-AddressInformation.xsl

```diff
@@ -0,0 +1,170 @@
+    <xsl:import href="../../../fhir/2_fhir_fhir_include.xsl"/>
+    <xsl:output method="xml" indent="yes"/>
+    <xsl:strip-space elements="*"/>
+
+    <xd:doc scope="stylesheet">
+        <xd:desc>Converts ada adresgegevens to FHIR resource conforming to profile
+            zib-AddressInformation</xd:desc>
+    </xd:doc>
+
+    <xd:doc>
+        <xd:desc>Unwrap adresgegevens_registratie element</xd:desc>
+    </xd:doc>
+    <xsl:template match="adresgegevens_registratie">
+        <xsl:apply-templates select="adresgegevens" mode="zib-AddressInformation"/>
+    </xsl:template>
+
+    <xd:doc>
+        <xd:desc>Produces FHIR Address datatypes with address elements.</xd:desc>
+        <xd:param name="in">Ada 'adresgegevens' element containing the zib data</xd:param>
+    </xd:doc>
+    <xsl:template match="adresgegevens" mode="zib-AddressInformation" name="zib-AddressInformation" as="element(f:address)*">
+        <xsl:param name="in" select="." as="element()*"/>
+        <xsl:for-each select="$in[.//@value]">
+            <xsl:variable name="lineItems" as="element()*">
+                <xsl:for-each select="straat/@value">
+                    <extension url="http://hl7.org/fhir/StructureDefinition/iso21090-ADXP-streetName">
+                        <valueString value="{normalize-space(.)}"/>
+                    </extension>
+                </xsl:for-each>
+                <xsl:for-each select="huisnummer/@value">
+                    <extension url="http://hl7.org/fhir/StructureDefinition/iso21090-ADXP-houseNumber">
+                        <valueString value="{normalize-space(.)}"/>
+                    </extension>
+                </xsl:for-each>
+                <xsl:for-each select="huisnummerletter/@value | huisnummertoevoeging/@value">
+                    <extension url="http://hl7.org/fhir/StructureDefinition/iso21090-ADXP-buildingNumberSuffix">
+                        <valueString value="{normalize-space(.)}"/>
+                    </extension>
+                </xsl:for-each>
+                <xsl:for-each select="additionele_informatie/@value">
+                    <extension url="http://hl7.org/fhir/StructureDefinition/iso21090-ADXP-unitID">
+                        <valueString value="{normalize-space(.)}"/>
+                    </extension>
+                </xsl:for-each>
+                <xsl:for-each select="aanduiding_bij_nummer/@value">
+                    <extension url="http://hl7.org/fhir/StructureDefinition/iso21090-ADXP-additionalLocator">
+                        <valueString value="{normalize-space(.)}"/>
+                    </extension>
+                </xsl:for-each>
+            </xsl:variable>
+            <address>
+                <!--adres_soort-->
+                <xsl:for-each select="adres_soort[@codeSystem = '2.16.840.1.113883.5.1119'][@code]">
+                    <xsl:choose>
+                        <!-- Postadres -->
```

Comment:
De herhaalde stukken:
```xml
                           <extension url="http://nictiz.nl/fhir/StructureDefinition/ext-zib-AddressInformation-AddressType">
                                <valueCodeableConcept>
                                    <coding>
                                        <system value="{normalize-space(adres_soort/@codeSystem)}"/>
                                        <code value="HV"/>
                                        <display value="Vakantie adres"/>
                                    </coding>
                                </valueCodeableConcept>
                            </extension>
```
kunnen volgens mij vervangen worden door eenmalig:
```xml
                           <extension url="http://nictiz.nl/fhir/StructureDefinition/ext-zib-AddressInformation-AddressType">
                                <valueCodeableConcept>
                                    <xsl:call-template name="code-to-CodeableConcept"/>
                                </valueCodeableConcept>
                            </extension>
```
voorafgaand aan de `<xsl:choose>`

Reply:
Converted to:

```
   <valueCodeableConcept>
                                    <xsl:call-template name="code-to-CodeableConcept">
                                        <xsl:with-param name="in" select="."/>
                                    </xsl:call-template>
                                </valueCodeableConcept>
```